lib/Compilation.js

```diff
@@ -832,64 +877,121 @@ class Compilation extends Tapable {
 					this.removeChunkFromDependencies(d.module, chunk);
 				}
 			}
+		};
+
+		var blocks = block.blocks;
+		for(var indexBlock = 0; indexBlock < blocks.length; indexBlock++) {
+			var chunks = blocks[indexBlock].chunks;
+			for(var indexChunk = 0; indexChunk < chunks.length; indexChunk++) {
+				var blockChunk = chunks[indexChunk];
+				chunk.removeChunk(blockChunk);
+				blockChunk.removeParent(chunk);
+				this.removeChunkFromDependencies(chunks, blockChunk);
```

Comment:
here makes a wrong refactoring. `removeChunkFromDependencies` accepts a DependenciesBlock as first argument rather than array of chunks. Before here used a block.

The mistake doesn't fail test because of there is no test covering this code. The code from L886-L889 is never enterred in all tests